test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs

```diff
@@ -2857,6 +2863,156 @@ public void PackageBuilderPreserveFileLastWriteTime()
             }
         }
 
+        [Fact]
+        public void PackageBuilderCorrectLastWriteTimeForZipfileBefore1980()
+        {
+            // https://github.com/NuGet/Home/issues/7001
+            // Act
+            DateTime Year2020Date = new DateTime(2020, 12, 14, 23, 59, 7, DateTimeKind.Utc);
+            DateTimeOffset lastWriteTime = ZipFormatMinDate.AddSeconds(-1);
+            int numberOfDateCorrectedFiles = 0;
+            int numberOfDateNotCorrectedFiles = 0;
+            TestLogger innerLogger = new TestLogger();
+            ILogger logger = new PackCollectorLogger(innerLogger, new WarningProperties());
+
+            using (var directory = new TestLastWriteTimeDirectory(lastWriteTime.LocalDateTime))
+            {
+                var builder = new PackageBuilder(logger) { Id = "test", Version = NuGetVersion.Parse("1.0"), Description = "test" };
+                builder.Authors.Add("test");
+
+                // Create a file that is modified after 01/01/1980 0:00:00 UTC (midnight) and it shouldn't be modified.
+                string after1980File1 = Path.Combine(directory.Path, "After1980.txt");
+                string after1980File2 = Path.Combine(directory.Path, "After1980_2.txt");
+                string after1980File3 = Path.Combine(directory.Path, "After1980_3.txt");
+                File.WriteAllText(after1980File1, string.Empty);
+                File.WriteAllText(after1980File2, string.Empty);
+                File.WriteAllText(after1980File3, string.Empty);
+                File.SetLastWriteTime(after1980File1, Year2020Date);
+                File.SetLastWriteTime(after1980File2, ZipFormatMinDate);
+                File.SetLastWriteTime(after1980File3, ZipFormatMinDate.AddSeconds(1));
+
+                builder.AddFiles(directory.Path, "**", "Content");
+
+                using (var stream = new MemoryStream())
+                {
+                    builder.Save(stream);
+
+                    // Assert
+                    using (var archive = new ZipArchive(stream, ZipArchiveMode.Read, leaveOpen: true))
+                    {
+                        foreach (ZipArchiveEntry entry in archive.Entries)
+                        {
+                            string path = Path.Combine(directory.Path, entry.Name);
+
+                            // Please note: ZipArchive stream reader randomly changes LastWriteTime by another 1 second off than what entry.LastWriteTime has, most likely bug on their what we specified here.
```

Comment:
Here is the example, ugly bug happens randomly spent quite a lot time to reason with it.

![image](https://user-images.githubusercontent.com/8766776/102398598-59d87600-3f94-11eb-8536-0f597d982f24.png)